pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -374,6 +374,8 @@ func (e *EndpointController) syncService(key string) {
 			tolerateUnreadyEndpoints = b
 		} else {
 			glog.Errorf("Failed to parse annotation %v: %v", TolerateUnreadyEndpointsAnnotation, err)
+			e.queue.Add(key) // Retry
```

Comment:
why retry when we fail to parse annotation? won't that cause a retry storm if someone sets the annotation wrong? i.e I can set: `service.alpha.kubernetes.io/tolerate-unready-endpoints=123` and I don't want it to retry because it's going to keep failing. When I modify the annotation to `service.alpha.kubernetes.io/tolerate-unready-endpoints=true`, we get a watch notification and it's enough to retry at that time.